apollo-ast/src/main/resources/apollo.graphqls

```diff
@@ -18,3 +18,9 @@ directive @typePolicy(keyFields: String!) on OBJECT
 # Indicates how to compute a key from a field arguments.
 # `keyArgs` should contain a selection set. Composite args are not supported yet.
 directive @fieldPolicy(forField: String!, keyArgs: String!) repeatable on OBJECT
+
+# Indicates that the given field or enum value is still experimental and might be changed
+# in a backward incompatible manner
+directive @experimental(
+  reason: String! = "Experimental"
+) on FIELD_DEFINITION | ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION | ENUM_VALUE
```

Comment:
There's a [graphql-spec work item for that](https://github.com/graphql/graphql-spec/pull/805/) 😅 

I can do this as part of a separate PR

Reply:
Oh! 😅. Well I guess it's not urgent until it's accepted but also wouldn't hurt to be ready in advance :)

Reply:
Just added `@deprecated` on input fields and field arguments. Arguments are an interesting case because I haven't found a way to "touch" them from Kotlin code. They are either GraphQL variables or values. I added validation rules so using them will emit a warning during codegen. But no annotation is emitted there.

Reply:
Very cool! 👍